apps/client/src/sidebar/pages/channel.tsx

```diff
@@ -1,35 +1,156 @@
+            >
+              {user.name}
+            </div>
+          </div>
+        );
+      })}
+    </div>
+  );
+};
+
+const ChannelMessage = ({
+  message,
+}: {
+  message: {
+    __typename?: "ChannelMessage" | undefined;
+    id: number;
+
+    author: {
+      __typename?: "User" | undefined;
+      id: number;
+      name: string;
+      avatar: string;
+    };
+    readBy: {
+      __typename?: "ChannelMessageRead" | undefined;
+      user: {
+        id: number;
+        name: string;
+        avatar: string;
+      };
+    }[];
+    content: string;
+    sentAt: number;
+  };
+}) => {
+  const navigate = useNavigate();
   const getDate = (time: number): Date => {
     return new Date(time);
   };
+  return (
+    <div className="">
+      <div className="mt-6 flex justify-center text-center text-xs text-slate-300">
+        {message.sentAt
+          ? getDate(+message.sentAt)
+              .toISOString()
+              .substring(0, 10) +
+            " - " +
+            getDate(+message.sentAt)
+              .toISOString()
+              .substring(11, 16)
+          : ""}
+      </div>
+      <div className="flex w-full flex-row">
+        <div className="flex w-9 shrink-0 justify-center">
+          <div className="flex justify-center self-end">
+            <img
+              className="flex h-6 w-6 basis-1 self-center rounded-full transition-all hover:h-7 hover:w-7"
+              src={message.author.avatar}
+              onClick={() => navigate(`/profile/${message.author.id}`)}
+            />
+          </div>
+        </div>
+        <div className="flex flex-col">
+          <div className="text-left text-xs tracking-wide text-slate-400">
+            <span>{message.author.name} </span>
+          </div>
+          <div className="rounded-md bg-slate-200 px-4 py-2 text-left tracking-wide">
+            {message.content}
+          </div>
+        </div>
+      </div>
+      <div className="flex flex-row ">
+        {
+          <ReadBy
+            users={message.readBy.map((Users) => {
+              return Users.user;
+            })}
+          />
+        }
+      </div>
+    </div>
+  );
+};
+
+export default function Channel() {
+  const { channelId } = useParams();
 
   if (!channelId) return <div>no channel id</div>;
 
   const { isLoading, isFetching, error, data } = useGetChannelQuery({
     channelId: +channelId,
   });
-
-  return (
-    <div className="flex flex-col">
-      <div className="mt-4 flex w-full flex-col items-center justify-center border-2 border-black p-2 text-center text-sm">
-        <div>Channel id: {channelId}</div>
-        <div>Channel: {data?.channel.name}</div>
-        <div>Owner: {data?.channel.owner.name}</div>
+  if (isLoading) {
+    return <div>Loading ...</div>;
+  }
+  if (isFetching) {
+    return <div>Fetching</div>;
+  }
+  if (error) {
+    return <div>Error</div>;
+  } else {
```

Comment:
We should probably have real loading and error page/components at some time

Reply:
Ok, I'm going to do this next